osu.Game/Overlays/Mods/ModPresetColumn.cs

```diff
@@ -75,9 +76,23 @@ private void asyncLoadPanels(IRealmCollection<ModPreset> presets, ChangeSet chan
                 Shear = Vector2.Zero
             }), loaded =>
             {
-                ItemsFlow.RemoveAll(panel => panel is ModPresetPanel);
+                removeAndDisposePresetPanels();
                 ItemsFlow.AddRange(loaded);
             }, (cancellationTokenSource = new CancellationTokenSource()).Token);
+
+            void removeAndDisposePresetPanels()
+            {
+                int i = 0;
+
+                while (i < ItemsFlow.Count)
```

Comment:
An alternative would be `foreachI (var p in temFlow.OfType<ModPresetPanel>.ToArray()))`.

Not sure which reads better.

Reply:
Sure, we can go with that too and save the index shenanigans for framework hot paths. Not hugely bothered either way myself in this context.